src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/SymbolTable.cs

```diff
@@ -898,11 +898,7 @@ private static List<object> BuildDeclarationChain(Type callingType)
                     ParameterInfo[] parameters = methodBase.GetParameters();
 
                     bool bAdded = false;
-#if UNSUPPORTEDAPI
-                    foreach (MethodInfo methinfo in Enumerable.Where(t.DeclaringType.GetRuntimeMethods(), m => m.MetadataToken == methodBase.MetadataToken))
-#else
                     foreach (MethodInfo methinfo in Enumerable.Where(t.DeclaringType.GetRuntimeMethods(), m => m.HasSameMetadataDefinitionAs(methodBase)))
```

Comment:
It looks like HasSameMetadataDefinitionAs checks more than just MetadataToken equality, but also Module equality, so switching to use HasSameMetadataDefinitionAs is potentially more strict than the UNSUPPORTEDAPI version?